engine/access/ingestion/engine.go

```diff
@@ -111,100 +159,204 @@ func New(
 		collectionsToMarkExecuted:  collectionsToMarkExecuted,
 		blocksToMarkExecuted:       blocksToMarkExecuted,
 		rpcEngine:                  rpcEngine,
+
+		// queue / notifier for execution receipts
+		executionReceiptsNotifier: engine.NewNotifier(),
+		executionReceiptsQueue:    executionReceiptsQueue,
+
+		// queue / notifier for finalized blocks
+		finalizedBlockNotifier: engine.NewNotifier(),
+		finalizedBlockQueue:    finalizedBlocksQueue,
+
+		messageHandler:            messageHandler,
+		processBackgroundDisabled: false,
 	}
 
+	// Add workers
+	e.ComponentManager = component.NewComponentManagerBuilder().
+		AddWorker(e.processBackground).
+		AddWorker(e.processExecutionReceipts).
+		AddWorker(e.processFinalizedBlocks).
+		Build()
+
 	// register engine with the execution receipt provider
-	_, err := net.Register(engine.ReceiveReceipts, eng)
+	_, err = net.Register(engine.ReceiveReceipts, e)
 	if err != nil {
 		return nil, fmt.Errorf("could not register for results: %w", err)
 	}
 
-	return eng, nil
+	return e, nil
 }
 
-// Ready returns a ready channel that is closed once the engine has fully
-// started. For the ingestion engine, we consider the engine up and running
-// upon syncing all the missing collections
-func (e *Engine) Ready() <-chan struct{} {
-	// request all the missing collection upfront
-	readyChan := e.unit.Ready(func() {
-		ctx, cancel := context.WithTimeout(context.Background(), defaultCollectionCatchupTimeout)
-		defer cancel()
-		err := e.requestMissingCollections(ctx)
-		if err != nil {
-			e.log.Error().Err(err).Msg("requesting missing collections failed")
-		}
-	})
-	e.unit.LaunchPeriodically(e.updateLastFullBlockReceivedIndex, defaultFullBlockUpdateInterval, time.Duration(0))
-	return readyChan
+func (e *Engine) DisableProcessBackground() {
+	// disable the processBackground worker from starting during testing to avoid calling mocks
+	e.processBackgroundDisabled = true
```

Comment:
It is better to avoid this, this isn't a good practice. The proper way to address this would be to setup the mocks correctly.